src/main/java/org/assertj/core/api/ErrorCollector.java

```diff
@@ -36,24 +39,28 @@
   private final LastResult lastResult = new LastResult();
 
   @RuntimeType
-  public Object intercept(@This Object obj, @SuperCall Callable<Object> proxy,
-                          @SuperMethod(nullIfImpossible = true) Method method) throws Exception {
+  public static Object intercept(
+    @FieldValue(FIELD_NAME) ErrorCollector errorCollector,
+    @This Object obj,
+    @SuperCall Callable<?> proxy,
+    @SuperMethod(nullIfImpossible = true) Method method,
+    @StubValue Object stub) throws Exception {
```

Comment:
Can you clarify why stub is needed here ? it's not clear to me (the previous code was not to clear neither to be fair)

Reply:
Byte Buddy provides an object for each argument whenever an intercepted method is invoked.

This: The instance of the method, the `this` reference.
SuperCall: A proxy to invoke the original method.
SuperMethod: A reference to the original method.
StubValue: A default value for the return type. null for reference type and 0 for the corresponding primitive types.